lib/Sema/CSSimplify.cpp

```diff
@@ -6975,7 +7111,8 @@ static bool isForKeyPathSubscriptWithoutLabel(ConstraintSystem &cs,
   if (auto *SE = getAsExpr<SubscriptExpr>(locator->getAnchor())) {
     auto *indexExpr = SE->getIndex();
     return isa<ParenExpr>(indexExpr) &&
-           isa<KeyPathExpr>(indexExpr->getSemanticsProvidingExpr());
+           (isa<KeyPathExpr>(indexExpr->getSemanticsProvidingExpr()) ||
+            isa<CodeCompletionExpr>(indexExpr->getSemanticsProvidingExpr()));
```

Comment:
I don’t think code completion expressions satisfy the requirement that they **are** for key paths subscripts. I think a better name for this function would now be `mayBeForKeyPathSubscriptWithoutLabel`. WDYT?